File: Java/Navigation/app/src/main/java/com/ob/navigation/SecondFragment.java
package com.ob.navigation;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.ob.navigation.databinding.FragmentFirstBinding;
import com.ob.navigation.databinding.FragmentSecondBinding;

public class SecondFragment extends Fragment
{
    FragmentSecondBinding binding;

    @Override
    public void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
    {
        binding = FragmentSecondBinding.inflate(inflater, container, false);
        View view = binding.getRoot();
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState)
    {
        super.onViewCreated(view, savedInstanceState);

        Button btnGoToBack = binding.btnGoToBack;

        btnGoToBack.setOnClickListener(view1 -> {
            NavDirections action = SecondFragmentDirections.actionSecondFragmentToFirstFragment();
            Navigation.findNavController(view1).navigate(action);
        });
    }

    @Override
    public void onDestroy()
    {
        super.onDestroy();
        binding = null;
    }
}

File: Kotlin/Navigation/app/src/main/java/com/ob/navigation/SecondFragment.kt
package com.ob.navigation

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.Navigation
import com.ob.navigation.databinding.FragmentFirstBinding
import com.ob.navigation.databinding.FragmentSecondBinding

class SecondFragment : Fragment() {

    private var _binding: FragmentSecondBinding? = null
    private val binding get() = _binding!!
    private var name = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        _binding = FragmentSecondBinding.inflate(inflater, container, false)
        val view = binding.root
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val btnGoToSecond = binding.btnGoToBack
        val textSecond = binding.textSecond

        arguments?.let {
            name = SecondFragmentArgs.fromBundle(it).name
            textSecond.setText(name)
        }

        btnGoToSecond.setOnClickListener {
            val action = SecondFragmentDirections.actionSecondFragmentToFirstFragment()
            Navigation.findNavController(it).navigate(action)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}